rfcs/2020-09-21-0000-kubernetes-metrics-overview.md

```diff
@@ -0,0 +1,283 @@
+# RFC 0000 - 2020-09-21 - Overview of metrics in Kubernetes
+
+This RFC's goal is to create a systematic overview of metrics in the
+Kubernetes ecosystem.
+
+## Table of contents
+
+- [RFC 3684 - 2020-09-21 - Overview of metrics in Kubernetes](#rfc-3684---2020-09-21---overview-of-metrics-in-kubernetes)
+  - [Table of contents](#table-of-contents)
+  - [Motivation](#motivation)
+  - [Overview](#overview)
+  - [Metrics exposed by the Kubernetes system components](#metrics-exposed-by-the-kubernetes-system-components)
+    - [Properties](#properties)
+    - [Per-component insights](#per-component-insights)
+  - [TODO](#todo)
+  - [Prior art](#prior-art)
+
+## Motivation
+
+Metrics is Kubernetes are a large sector, with many concerns. We need a way to
+better understand it, to, sort of, have a list of entry points to explore in
+more detail.
+
+This overview will help us reason about the metrics  in Kubernetes and make
+educated design and planning decisions.
+
+## Overview
+
+- Application metrics exposed via [OpenMetrics](https://openmetrics.io/) or
+  [Prometheus exposition format](https://prometheus.io/docs/instrumenting/exposition_formats/)
+  for scraping, aka pull-based gathering.
+
+- Application metrics shipped to a server of some kind, typically via `statsd`
+  protocol, aka push-based gathering.
+
+  In the modern metrics aggregation stacks, these are usually converted to
+  pull-based gathering via
+  [pushgateway](https://github.com/prometheus/pushgateway) or
+  [statsd_exporter](https://github.com/prometheus/statsd_exporter) - but
+  it's even more common to not have any push-based metrics at all.
+
+- Metrics exposed by the Kubernetes system components via Prometheus exposition
```

Comment:
If we scrape Prometheus itself, as described in https://github.com/timberio/vector/pull/4098/files#r495493261, we wouldn't need to deal with any of this, right? The Prometheus/K8s ecosystem handles gluing all of this together. The end result is that all of these metrics are stored in Prometheus itself, correct?

Reply:
Yes, exactly!